community/plugins/weather-query-plugin-starter/src/main/java/com/alibaba/cloud/ai/plugin/GetWeatherService.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.alibaba.cloud.ai.plugin;
+
+import com.fasterxml.jackson.annotation.JsonClassDescription;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.annotation.JsonPropertyDescription;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.HttpHeaders;
+import org.springframework.util.StringUtils;
+import org.springframework.web.reactive.function.client.WebClient;
+import org.springframework.web.util.UriComponentsBuilder;
+import reactor.core.publisher.Mono;
+
+import java.util.Map;
+import java.util.function.Function;
+
+public class GetWeatherService
+        implements Function<GetWeatherService.Request, GetWeatherService.Response> {
+
+    private static final Logger logger = LoggerFactory.getLogger(GetWeatherService.class);
+
+    private static final String WEATHER_API_URL = "https://api.weatherapi.com/v1/current.json";
+
+    private final WebClient webClient;
+
+    private final ObjectMapper objectMapper = new ObjectMapper();
+
+    @Value("${spring.ai.alibaba.plugin.weather.api-key}")

```

Comment:
pls see https://github.com/alibaba/spring-ai-alibaba/pull/112#discussion_r1849404070

configuration can be injected like this